test/vni-manager-mocha.js

```diff
@@ -0,0 +1,377 @@
+/**
+ * File: vni-manager-mocha.js
+ * Unit tests for the vni APIs defined in src/vni-manager.js
+ */
+
+var chai = require('chai');
+var expect = chai.expect;
+var should = chai.should();
+
+var componentFactory = require('../src/noflo-component-factory');
+var createState = require('../src/create-state');
+var inputStates = require('../src/input-states');
+var promiseOutput = require('../src/promise-output');
+
+var test = require('./common-test');
+
+var vniManager = require('../src/vni-manager');
+
+describe("vni-manager", function() {
+
+    beforeEach(function() {
+
+        node = test.createComponent(componentFactory({
+            inPorts:{input:{
+                ondata: promiseOutput(function(payload){
+                    return payload + " world";
+                })
+            }},
+            outPorts: promiseOutput.outPorts
+          }, vniManager )
+        );
+
+    });
+
+    afterEach(function() {
+        // clean up
+        node.deleteAllVnis();
+    });
+
+    it("should exist as a function", function() {
+        vniManager.should.exist;
+        vniManager.should.be.a('function');
+    });
+
+    it("should return a component facade", function() {
```

Comment:
My take away from yesterdays discussion on this is that the term node should be used when referencing a particular instance and the term component should be used when referencing a class/collection of them.